### DISCUSSION TITLE
There is a way to create a route that opens a popup in my app?

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [vercel/next.js#17687] [Help] [answered]
Hi Guys, 
with react-router-dom, our login has link, but it opens on a popup, there is a way to do the same with nextjs?

I need that "app.com/login" opens a popup with the form on the screen.

Thanks so much!

**Comment by jamesmosier** [comment, marked as answer]
Hi @wilsonneto-dev. Check out the [with-route-as-modal](https://github.com/vercel/next.js/tree/canary/examples/with-route-as-modal) example. It shows how to open a modal depending on what route the user is on.

**Reply by mk0y** [reply]
This forces you to use a certain library.

**Comment by seitbekir** [comment]
You may use [Parallel Routes](https://nextjs.org/docs/app/building-your-application/routing/parallel-routes).

Just make a socket to place a modal and call you modal via `<Link>`.

```tsx
// layout.tsx
export default function Layout({ children, popup }: {
  children: ReactNode;
  popup: ReactNode;
}) {
  return (
    <div>
      {children}
      {popup}
    </div>
  );
}
```

and for the modal use

```tsx
// @something-you-need/page.tsx
export default function Page() {
  const router = useRouter();

  const close = () => router.push("/");
  return (
    <Popup>...</Popup>
  );
}
```
